### PR TITLE
[Backport 6.2] HSEARCH-4947 Make sure that nested jars can be read on Windows

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-4947

Backport of #3730 to branch 6.2

I've added the schema export test updates too, so it'll be easier to backport things in the future 😃

### CODE REVIEW

File: integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/schema/management/ElasticsearchSchemaManagerExporterIT.java [hunk 1]
```diff
@@ -35,7 +38,10 @@ public class ElasticsearchSchemaManagerExporterIT {
 	@Rule
 	public final TemporaryFolder temporaryFolder = new TemporaryFolder();
 	@Rule
-	public OrmSetupHelper setupHelper = OrmSetupHelper.withSingleBackend( BackendConfigurations.simple() );
+	public OrmSetupHelper setupHelper = OrmSetupHelper.withMultipleBackends(
+			BackendConfigurations.simple(),
+			Map.of( Article.BACKEND_NAME, BackendConfigurations.simple() )
```
Comment (yrodiere) [inline integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/schema/management/ElasticsearchSchemaManagerExporterIT.java:43]: This won't work on JDK 8, will it?

File: integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/schema/management/LuceneSchemaManagerExporterIT.java [hunk 2]
```diff
@@ -31,17 +32,16 @@ public class LuceneSchemaManagerExporterIT {
 	@Rule
 	public final TemporaryFolder temporaryFolder = new TemporaryFolder();
 	@Rule
-	public OrmSetupHelper setupHelper = OrmSetupHelper.withSingleBackend( BackendConfigurations.simple() );
+	public OrmSetupHelper setupHelper = OrmSetupHelper.withMultipleBackends(
+			BackendConfigurations.simple(),
+			Map.of( Article.BACKEND_NAME, BackendConfigurations.simple() )
```
Comment (yrodiere) [inline integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/schema/management/LuceneSchemaManagerExporterIT.java:37]: Same, won't work on JDK 8.

**Review by yrodiere — CHANGES_REQUESTED** [review]

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
> I've added the schema export test updates too, so it'll be easier to backport things in the future 😃

@marko-bekhta Looks like something is wrong with this

**Comment by marko-bekhta** [comment]
😔 jakarta/orm6 modules have their own poms ... so tests were' not excluded from the multi-backend runs for them...

**Review by yrodiere — APPROVED** [review]

**Pull request merged**